File: samples/Test_JS/Resources/js/Control/TabBarTest.js
/**
 * Created by crossApp on 16/8/3.
 */

var TabBarControllerTest = ca.CAViewController.extend({

    ctor: function () {
        this._super();
    },

    viewDidLoad: function() {

        var item1 = ca.CATabBarItem.create("item1", ca.CAImage.create(""), ca.CAImage.create(""));
        var item2 = ca.CATabBarItem.create("item2", ca.CAImage.create(""), ca.CAImage.create(""));
        var item3 = ca.CATabBarItem.create("item3", ca.CAImage.create(""), ca.CAImage.create(""));

        var item = new Array;
        item.push(item1);
        item.push(item2);
        item.push(item3);

        var tabBar = ca.CATabBar.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout.fill(), ca.DVerticalLayout_H_C(98,0.33)));
        tabBar.setItems(item);
        tabBar.showSelectedIndicator();
        tabBar.setSelectedAtIndex(1);
        this.getView().addSubview(tabBar);
    },
});

File: samples/Test/Classes/Control/TabBarTest.cpp

#include "TabBarTest.h"
#include "FirstViewController.h"

TabBarTest::TabBarTest()
{
    this->setTitle("CATabBar");
}

TabBarTest::~TabBarTest()
{
}

void TabBarTest::viewDidLoad()
{
    CATabBarItem* item1 = CATabBarItem::create("item1", CAImage::create(""), CAImage::create(""));
    CATabBarItem* item2 = CATabBarItem::create("item2", CAImage::create(""), CAImage::create(""));
    CATabBarItem* item3 = CATabBarItem::create("item3", CAImage::create(""), CAImage::create(""));

    CAVector<CATabBarItem*> item;
    item.pushBack(item1);
    item.pushBack(item2);
    item.pushBack(item3);
    
    CATabBar* tabBar = CATabBar::createWithLayout(DLayout(DHorizontalLayoutFill, DVerticalLayout_H_C(98, 0.33)));
    tabBar->setItems(item);
    tabBar->showSelectedIndicator();
    tabBar->setSelectedAtIndex(0);
    tabBar->setTag(100);
    
    this->getView()->addSubview(tabBar);
}

void TabBarTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

